homeassistant/components/entur_public_transport/sensor.py

```diff
@@ -21,8 +21,10 @@
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 from homeassistant.util import Throttle
 import homeassistant.util.dt as dt_util
+from homeassistant.core import Context
 
-API_CLIENT_NAME = "homeassistant-homeassistant"
+# Entur API has ratelimiting per client ID. We therefore need a unique ID per install
+API_CLIENT_NAME = "homeassistant-{}".format(Context(user_id=user.id))
```

Comment:
Context is not really a fit here. The context object is used when things trigger.
Also, according to upstream documentation, this must be in `<company>-<application>` format.

../Frenck

Reply:
Thank you for the feedback. I just tried to illustrate the idea of what I want to accomplish. Upstream documentation is targeted towards public transport companies in Norway. I have spoken to the developers and they want "private" usecases like Home Assistant instances to uniquely identify per user. Do you have a tip or best practice on how to achieve that here?

Reply:
Christian, did they say anything specific about what the unique identification should look like? I mean, should it be something identifiable, like an email address, or is something like a UUID OK?

Apparently, HA doesn't currently have a per-site UUID. As I suggested a couple of comments above, the zone.home entity could be used, e.g. by hashing the latitude and longitude attributes. This is assuming users set their Home Zone, which I expect most will, since it provides useful functionality.

Alternately, just require a configuration parameter, with some guidance in the documentation on what is required.

Reply:
@pvarlien They are not that strict about it. If it starts with e.g. "homeassistant-" the suffix part can be set by us as we want as long as it in some way is unique for each user/instance.